test/support/factory.ex

```diff
@@ -166,6 +167,72 @@ defmodule Wanda.Factory do
     }
   end
 
+  def completed_with_errors(
+        %Execution{execution_id: execution_id, group_id: group_id} = execution
+      ) do
+    result = %Result{
+      execution_id: execution_id,
+      group_id: group_id,
+      check_results: [
+        %CheckResult{
+          check_id: "some_check_id",
+          expectation_results: [
+            %ExpectationResult{
+              name: "some_expectation_name",
+              result: false,
+              type: :expect
+            }
+          ],
+          agents_check_results: [
+            %AgentCheckResult{
+              agent_id: UUID.uuid4(),
+              expectation_evaluations: [
+                %ExpectationEvaluation{
+                  name: "some_expectation_name",
+                  return_value: true,
+                  type: :expect
+                },
+                %ExpectationEvaluationError{
+                  name: "some_expectation_name",
+                  message: Faker.StarWars.quote(),
+                  type: :fact_missing_error
+                }
+              ]
+            },
+            %AgentCheckError{
```

Comment:
It would be nice to have a new factory for `AgentCheckError` so we can reuse it here